### PR TITLE
Add a new option to control element comparison

### PR DESCRIPTION
**Pull request by theKashey**
Just adding an option (to pair `vnode`) to hook element comparison. Required by #1007 
Another option is to use @arqex's code - https://github.com/developit/preact/compare/master...arqex:master  with setComponentComparator function.  It is just a matter of taste.

Looking forward to make Preact quite 🔥[loadable]
(😕I did not found tests for vnode, and did not add my ones next to them  😭)

### CODE REVIEW

**Comment by theKashey** [comment]
New version of React-Hot-Loader with partial Preact support was released.
Waiting for this stuff for full compliance.

File: src/vdom/index.js [hunk 1]
```diff
@@ -1,5 +1,18 @@
...
 
+/**
+ * Checks that two components are equal
+ * @param {Component} component1 First component to compare
+ * @param {Component} component2 Second component to compare
+ * @return {boolean}
+ * @private
+ */
+export function areComponentsEqual(component1, component2) {
+	return options.areComponentsEqual
```
Comment (marvinhagemeister) [inline src/vdom/index.js]: One could remove the `if` check here by making it the default implementation for `options.areComponentsEqual`

Comment (theKashey) [reply src/vdom/index.js]: Pro - speed matters
Cons - you will be unable to revert configuration. Not a big deal, but could be nasty.

Comment (theKashey) [reply src/vdom/index.js]: Probably one `if` here is way faster than one more function call you are asking for.

Comment (theKashey) [reply src/vdom/index.js]: @marvinhagemeister - you were right.
My variant is 4x times slower than yours, and yours are equal to the base line.

https://jsperf.com/call-or-if/1

**Comment by theKashey** [comment]
Obsolete

**Pull request closed without merging**